### PR TITLE
Implementation proposal for #60

### PR DESCRIPTION
**Pull request by PuKoren**

### CODE REVIEW

**Comment by PuKoren** [comment]
Please review again following changes

File: src/index.js [hunk 1]
```diff
@@ -47,9 +46,7 @@ function Carotte(config) {
         deadLetterQualifier: 'dead-letter',
         enableDeadLetter: true,
         autoDescribe: false,
-        retryOnError() {
-            return true;
-        },
+        retryOnError() { },
```
Comment (nbeydoon) [inline src/index.js]: Not useful anymore

File: src/index.js [hunk 2]
```diff
@@ -79,28 +76,20 @@ function Carotte(config) {
...
 
         connexion = amqp.connect(`amqp://${config.host}`, config.connexion).then(conn => {
-            conn.on('close', error => {
-                config.transport.error('amqp.connection.closed', { error });
-                connexion = null;
+            conn.on('close', (err) => {
+                connexion = undefined;
                 channels = {};
                 carotte.cleanExchangeCache();
+                carotte.onClose(err);
```
Comment (nbeydoon) [inline src/index.js]: Maybe an error because onClose = onError at line:607

File: src/index.js [hunk 3]
```diff
@@ -79,28 +76,20 @@ function Carotte(config) {
         }
 
         connexion = amqp.connect(`amqp://${config.host}`, config.connexion).then(conn => {
-            conn.on('close', error => {
-                config.transport.error('amqp.connection.closed', { error });
-                connexion = null;
+            conn.on('close', (err) => {
+                connexion = undefined;
```
Comment (nbeydoon) [inline src/index.js]: we lose the event name "amqp.connection.closed" if we only use error

File: src/index.js [hunk 4]
```diff
@@ -79,28 +76,20 @@ function Carotte(config) {
...
                 channels = {};
                 carotte.cleanExchangeCache();
+                carotte.onClose(err);
             });
-            conn.once('error', error => {
-                config.transport.error('amqp.connection.error', { error });
-                connexion = null;
-                channels = {};
-            });
+            conn.once('error', carotte.onError);
```
Comment (nbeydoon) [inline src/index.js]: same here we lose amqp.connection.error

**Pull request merged**